src/plugins/embeddable/public/lib/filterable_embeddable/should_fetch.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fastIsEqual from 'fast-deep-equal';
+import { Observable } from 'rxjs';
+import { map, distinctUntilChanged, skip, startWith } from 'rxjs/operators';
+import type { Filter, Query, TimeRange } from '@kbn/es-query';
+import { COMPARE_ALL_OPTIONS, onlyDisabledFiltersChanged } from '@kbn/es-query';
+import { EmbeddableInput } from '../embeddables';
+
+export const shouldRefreshFilterCompareOptions = {
+  ...COMPARE_ALL_OPTIONS,
+  // do not compare $state to avoid refreshing when filter is pinned/unpinned (which does not impact results)
+  state: false,
+};
+
+type FilterableEmbeddableInput = EmbeddableInput & {
```

Comment:
I wonder if this should be exported and moved to the types file?

There are a lot of instances where embeddables redefine these input keys. Maybe in a followup we could switch them to use something like `FilterableEmbeddableInput & { ...other input }`

Reply:
resolved with https://github.com/elastic/kibana/pull/151799/commits/dfa8831f159d5ddf4ae6e3c9af24cc3c931b6a0b